src/VisionNode.cpp

```diff
@@ -334,33 +334,31 @@ int main(int argc, char **argv)
     // Queue and callback for collecting images
     std::queue<sensor_msgs::Image::ConstPtr> message_queue;
 
-    std::function<void(const sensor_msgs::Image::ConstPtr&)> handler =
+    std::function<void(const sensor_msgs::Image::ConstPtr&)> image_msg_handler =
         [&](const sensor_msgs::Image::ConstPtr& message) {
             message_queue.push(message);
         };
 
     image_transport::ImageTransport image_transporter{nh};
-    ros::Subscriber sub = nh.subscribe(
+    image_transport::Subscriber sub = image_transporter.subscribe(
         "/bottom_image_raw/image_raw",
         100,
-        &std::function<void(const sensor_msgs::Image::ConstPtr&)>::operator(),
-        &handler);
+        image_msg_handler);
 
     // Main loop
     while (ros::ok())
     {
         if (!message_queue.empty()) {
-            if (message_queue.size() > message_queue_item_limit) {
+            sensor_msgs::Image::ConstPtr message = message_queue.front();
+            message_queue.pop();
+
+            if (message_queue.size() >= message_queue_item_limit) {
```

Comment:
I think this logic is unclear. I think the intention is more clear if you do message_queue_item_limit - 1 because you did the pop in the line above.

Reply:
Fixed

Reply:
Your fix is now not equivalent. You changed it to `if (message_queue.size() > message_queue_item_limit - 1)` when shouldn't it be  `if (message_queue.size() >= message_queue_item_limit - 1)`?

Reply:
I think the goal was to keep the functionality the same but make it clearer?  So it should have the same effect as the previous version

Reply:
Yes but x>=5 is different than x>4. It adds 4 and 5 to the set that makes the condition true when didnt you only want to add 5?

Reply:
No either way it's true when x is 5 and false when x is 4

Reply:
I think I’m confusing myself and that you’re right.